py3status/modules/deadbeef.py

```diff
@@ -3,95 +3,101 @@
-            'full_text': ''
+    class Meta:
+        deprecated = {
+            'rename_placeholder': [
+                {
+                    'placeholder': 'elapsed',
+                    'new': 'playback_time',
+                    'format_strings': ['format'],
+                },
+            ],
         }
-        return response
 
-    # return track currently playing in deadbeef
-    def get_status(self):
+    def _is_running(self):
         try:
-            # check if we have deadbeef running
-            check_output(['pidof', 'deadbeef'])
-        except CalledProcessError:
-            return self._empty_response()
+            self.py3.command_output(['pgrep', 'deadbeef'])
+            return True
+        except:
+            return False
 
-        try:
-            # get all properties using ¥ as delimiter
-            status = check_output(['deadbeef',
-                                   '--nowplaying',
-                                   self.delimiter.join(['%a',
-                                                        '%t',
-                                                        '%l',
-                                                        '%e',
-                                                        '%y',
-                                                        '%n'])])
-
-            if status == 'nothing':
-                return self._empty_response()
-
-            # split properties using special delimiter
-            parts = status.split(self.delimiter)
-            if len(parts) == 6:
-                artist, title, length, elapsed, year, tracknum = parts
-            else:
-                return self._error_response(self.py3.COLOR_BAD)
-
-            response = {
+    def deadbeef(self):
+        color = self.py3.COLOR_BAD
+        artist = isplaying = length = playback_time = title = \
+            tracknumber = year = ""
+
+        if not self._is_running():
+            return {
                 'cached_until': self.py3.time_in(self.cache_timeout),
-                'full_text': self.py3.safe_format(self.format,
-                                                  dict(artist=artist,
-                                                       title=title,
-                                                       length=length,
-                                                       elapsed=elapsed,
-                                                       year=year,
-                                                       tracknum=tracknum)
-                                                  )
+                'color': color,
+                'full_text': '',
             }
-            return response
-        except:
-            return self._error_response(self.py3.COLOR_BAD)
+
+        # mix metadata fields in with the delimiters
+        fmt = self.delimiter.join([
+            '%artist%', '%isplaying%', '%length%', '%playback_time%',
+            '%title%', '%tracknumber%', '%year%'
+        ])
```

Comment:
I can't find any info that deadbeef supports Foobar2000 but ok.  So the real question is what does 

`deadbeef --nowplaying-tf %missing_param%` do because that really tells us the way forward.

I see no reason why we shouldn't support all sensible params eg `%whatever%`